### PR TITLE
Fixes issue #259: Updating validation to allow spaces in service and …

### PR DESCRIPTION
**Pull request by splatteredbits**
I removed spaces from the set of disallowed characters in the Topshelf.HostConfigurators.HostConfiguratorImpl's Validate method.

### CODE REVIEW

File: src/Topshelf.Tests/CommandLine_Specs.cs [hunk 1]
```diff
@@ -119,23 +119,39 @@ public void Should_create_an_install_host_with_description()
...
-
-            Assert.IsInstanceOf<InstallHost>(host);
-            var installHost = (InstallHost)host;
-            Assert.AreEqual("Joe", installHost.Settings.Name);
-            Assert.AreEqual("42", installHost.Settings.InstanceName);
-            Assert.AreEqual("Joe$42", installHost.Settings.ServiceName);
-        }
-
-        [Test]
+		[Test]
```
Comment (TravisTheTechie) [inline src/Topshelf.Tests/CommandLine_Specs.cs]: Whitespace should align, it looks like you're using 8 characters or tabs. Should be using 4 spaces per indent.

**Comment by TravisTheTechie** [comment]
I feel like there was a great reason we didn't support whitespaces in service names, but it's been long enough I can't remember what it was. :+1:

**Comment by splatteredbits** [comment]
OK. I've cleaned up the whitespace.

**Comment by splatteredbits** [comment]
I think I figured out why you didn't: your command line argument parser doesn't support them. It parses

    service.exe install -servicename:"Service Name"

As receiving these arguments:

 * -servicename:"Service
 * Name"

Looks like I also need to make the command line parser a little more sophisticated.

**Comment by TravisTheTechie** [comment]
`service install "-servicename:Service Name"`

**Comment by splatteredbits** [comment]
In that case, the arguments passed to TopShelf are:

 * "-servicename:Service
 * Name"

The quotes don't get stripped out by the shell.

**Comment by TravisTheTechie** [comment]
That doesn't seem right at all -- what shell & OS version are you using?

**Comment by splatteredbits** [comment]
I was puzzled as well. Windows 7, cmd.exe and PowerShell 5.

**Comment by TravisTheTechie** [comment]
I'm a little swamped this week, but I'll try and take a look. If not, maybe I can con @phatboyg into getting to it faster than I can.

**Comment by splatteredbits** [comment]
What's strange is from the command line, the `:` is required between names and values. In the unit tests, the colon is not required. On the command line this fails:

    service.exe install -servicename ServiceName

But this succeeds:

    service.exe install -servicename:ServiceName

**Comment by splatteredbits** [comment]
Seems the "problem" is in `Topshelf.CommandLineParser.CommandLine`'s `GetUnparsedCommandLine()` method. It returns the raw command line and skips .NET's parsing.

            string line = Environment.CommandLine;

            string applicationPath = Environment.GetCommandLineArgs().First();

            if (line == applicationPath)
                return "";

            if (line.Substring(0, applicationPath.Length) == applicationPath)
                return line.Substring(applicationPath.Length);

            string quotedApplicationPath = "\"" + applicationPath + "\"";

            if (line.Substring(0, quotedApplicationPath.Length) == quotedApplicationPath)
                return line.Substring(quotedApplicationPath.Length);

            return line;

Any reason you wrote your own parser instead of using .NET's?

**Comment by splatteredbits** [comment]
Figured out the quoting issue. Without colons between arg name and value, the value has to be in quotes. So this:

    service.exe install -servicename ServiceName

Should be:

    service.exe install -servicename "ServiceName"

**Comment by phatboyg** [comment]
Yep. Pretty sure that's documented in the help as well. 

____
Chris Patterson

**Comment by msell** [comment]
I can't seem to get a topshelf service installed with spaces in the name.  Is this pull request going to get accepted to correct the issue?

**Comment by phatboyg** [comment]
Merged.

**Pull request closed without merging**